ext/curl/curl.stub.php

```diff
@@ -26,6 +26,27 @@ final class CurlShareHandle
 {
 }
 
+#if LIBCURL_VERSION_NUM >= 0x073E00 /* Available since 7.62.0 */
+/**
+ * @strict-properties
+ * @not-serializable
+ */
+final class CurlUrl implements Stringable
+{
+    public function __construct(?string $url = null) {}
+
+    public function get(int $part, int $flags = 0): string {}
+
+    public function set(int $part, string $content, int $flags = 0): void {}
```

Comment:
I would love and would strongly suggest the usage of a wither-style, immutable API. We already have a `Date` class where mutability proved to be harmful, and I think an URL value object would be the very same use-case.
```suggestion
    public function with(CurlUrlPart $part, string $content, int $flags = 0): CurlUrl {}
```

Reply:
In the context of `CurlUrl` I would not make it `Immutable`. It's perfectly valid to do something like this : 

```php
$url = new CurlUrl('https://www.php.net');
$ch = curl_init();
curl_setopt($ch, CURLOPT_CURLU, $url);
curl_exec($ch);

$url->set('/downloads'); 
curl_exec($ch); 
```

But your point is still valid. It would be nice to have something like `ImmutableCurlUrl` so that user can choose one or the other depending on their context.